ui/components/src/model/units/percent.ts

```diff
@@ -43,32 +44,24 @@ export const PERCENT_UNIT_CONFIG: Readonly<Record<PercentUnitKind, UnitConfig>>
   },
 };
 
-const MAX_SIGNIFICANT_DIGITS = 3;
-
 export function formatPercent(value: number, { kind, decimal_places }: PercentUnitOptions): string {
   // Intl.NumberFormat translates 0 -> 0%, 0.5 -> 50%, 1 -> 100%
   if (kind === 'Percent') {
     value = value / 100;
   }
 
-  const hasDecimalPlaces = decimal_places !== undefined;
+  const formatterOptions: Intl.NumberFormatOptions = {
+    style: 'percent',
+    useGrouping: true,
+  };
 
-  let formatter;
+  const hasDecimalPlaces = decimal_places !== undefined && decimal_places !== null;
   if (hasDecimalPlaces) {
-    // If there is a specified # of decimal places, use maximumFractionDigits
-    formatter = new Intl.NumberFormat('en-US', {
-      style: 'percent',
-      maximumFractionDigits: decimal_places,
-      useGrouping: true,
-    });
+    formatterOptions.maximumFractionDigits = limitDecimalPlaces(decimal_places);
   } else {
-    // By default, use maximumSignificantDigits
-    formatter = new Intl.NumberFormat('en-US', {
-      style: 'percent',
-      maximumSignificantDigits: MAX_SIGNIFICANT_DIGITS,
-      useGrouping: true,
-    });
+    formatterOptions.maximumSignificantDigits = MAX_SIGNIFICANT_DIGITS;
```

Comment:
I've played a lot of code golf trying to make the formatting code readable. I think this is pretty nice 🤷‍♀️ Maybe tomorrow I'll change my mind :) 

But anyway, on the last PR, people weren't sure what the difference was between the formatting when `hasDecimalPlaces` is `true` and `hasDecimalPlaces` is `false`, so I thought this would make it super clear.